source/sessions/snapshot-sessions.rst

```diff
@@ -0,0 +1,358 @@
+================================
+Snapshot Reads Specification
+================================
+
+:Spec Title: Snapshot Reads Specification (See the registry of specs)
+:Spec Version: 1.0
+:Author: Boris Dogadov
+:Advisors: Jeff Yemin, A. Jesse Jiryu Davis, Judah Schvimer
+:Status: Draft (Could be Draft, Accepted, Rejected, Final, or Replaced)
+:Type: Standards
+:Minimum Server Version: 5.0 (The minimum server version this spec applies to)
+:Last Modified: 15-Jun-2021
+
+.. contents::
+
+--------
+
+Abstract
+========
+
+Version 5.0 of the server introduces support for readConcern level "snapshot" (non-speculative)
+for read commands outside of transactions, including on secondaries.
+This spec builds upon the Sessions Specification to define how an application
+requests "snapshot" level readConcern and how a driver interacts with the server
+to implement snapshot reads.
+
+Definitions
+===========
+
+META
+----
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”,
+“SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be
+interpreted as described in `RFC 2119 <https://www.ietf.org/rfc/rfc2119.txt>`_.
+
+Terms
+-----
+
+ClientSession
+    The driver object representing a client session and the operations that can be
+    performed on it.
+
+MongoClient
+    The root object of a driver's API. MAY be named differently in some drivers.
+
+MongoCollection
+    The driver object representing a collection and the operations that can be
+    performed on it. MAY be named differently in some drivers.
+
+MongoDatabase
+    The driver object representing a database and the operations that can be
+    performed on it. MAY be named differently in some drivers.
+
+ServerSession
+    The driver object representing a server session.
+
+Session
+    A session is an abstract concept that represents a set of sequential
+    operations executed by an application that are related in some way. This
+    specification defines how sessions are used to implement snapshot reads.
+
+Snapshot reads
+    Reads with readconcern level ``snapshot`` that occur outside of transactions on
+    both the primary and secondary nodes, including in sharded clusters.
+
+Snapshot cluster time
```

Comment:
How about "Snapshot timestamp"? It keeps confusing everyone that the param is called "atClusterTime" but its type is just Timestamp, not the signed cluster time document that we usually call a "cluster time". So I propose calling this a timestamp.

Reply:
Thanks, `Snapshot timestamp` sounds more natural.